### PR TITLE
fix: #15968 update to show booking question variables in Available variables

### LINKED ISSUE #15968: [CAL-4076] Under available variables for custom event, we should inform about availability of booking question responses as well
Show more 
image

e.g. for custom event name
image

This is what it looks like after booking
image

owner1-acme@example.com was the booker/scheduler email. It is the value of {email} variable provided by Booking Questions. Any custom booking question data is also available for use

<sub>[CAL-4076](https://linear.app/calcom/issue/CAL-4076/under-available-variables-for-custom-event-we-should-inform-about)</sub>

### PR DESCRIPTION
**Pull request by vijayraghav-io**
## What does this PR do?

- Fixes #15968 
- Fixes CAL-4076

https://www.loom.com/share/25d59e2ae76a465d8d6735990f61007f?sid=cc622ca4-c0c6-4025-a320-8e447f25b795

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have added a Docs issue [here](https://github.com/calcom/docs/issues/new) if this PR makes changes that would require a [documentation change](https://docs.cal.com). If N/A, write N/A here and check the checkbox.
- [x] N/A I confirm automated tests are in place that prove my fix is effective or that my feature works.

### CODE REVIEW

File: apps/web/components/eventtype/CustomEventTypeModal.tsx [hunk 1]
```diff
@@ -61,37 +61,49 @@ const CustomEventTypeModalForm: FC<CustomEventTypeModalFormProps> = (props) => {
...
-            <p className="text-emphasis">{t("location_info")}</p>
+            {event.bookingFields && (
+              <p className="text-subtle mb-2 ml-1 font-medium">{t("booking_question_response_variables")}</p>
+            )}
+            {event.bookingFields &&
+              Object.keys(event.bookingFields).map((bookingfield) => (
+                <div className="mb-2.5 flex font-normal">
+                  <p className="text-subtle ml-1 mr-5 w-32">{`{${bookingfield}}`}</p>
+                  <p className="text-emphasis">{event.bookingFields?.[bookingfield]?.toString()}</p>
+                </div>
```
Comment (anikdhabal) [inline apps/web/components/eventtype/CustomEventTypeModal.tsx:105]: pls add a key prop here

**Review by anikdhabal — APPROVED** [review]
LGTM! Nice work @vijayraghav-io

File: apps/web/components/eventtype/CustomEventTypeModal.tsx (hunk 1 above)
Comment (vijayraghav-io) [reply apps/web/components/eventtype/CustomEventTypeModal.tsx]: Ok will add

**Comment by vijayraghav-io** [comment]
> LGTM! Nice work @vijayraghav-io

Thankyou! @anikdhabal

File: apps/web/components/eventtype/CustomEventTypeModal.tsx (hunk 1 above)
Comment (anikdhabal) [reply apps/web/components/eventtype/CustomEventTypeModal.tsx]: I've added it

**Pull request merged**